solr/core/src/java/org/apache/solr/handler/admin/api/BackupCoreAPI.java

```diff
@@ -0,0 +1,179 @@
+                      : null;
+              BackupFilePaths incBackupFiles = new BackupFilePaths(repository, locationUri);
+              IncrementalShardBackup incSnapShooter =
+                  new IncrementalShardBackup(
+                      repository,
+                      core,
+                      incBackupFiles,
+                      prevShardBackupId,
+                      shardBackupId,
+                      Optional.ofNullable(backupCoreRequestBody.commitName));
+              return incSnapShooter.backup();
+            } else {
+              SnapShooter snapShooter =
+                  new SnapShooter(
+                      repository, core, locationUri, name, backupCoreRequestBody.commitName);
+              // validateCreateSnapshot will create parent dirs instead of throw; that choice is
+              // dubious.
+              // But we want to throw. One reason is that this dir really should, in fact must,
+              // already
+              // exist here if triggered via a collection backup on a shared file system. Otherwise,
+              // perhaps the FS location isn't shared -- we want an error.
+              if (!snapShooter.getBackupRepository().exists(snapShooter.getLocation())) {
+                throw new SolrException(
+                    SolrException.ErrorCode.BAD_REQUEST,
+                    "Directory to contain snapshots doesn't exist: "
+                        + snapShooter.getLocation()
+                        + ". "
+                        + "Note that Backup/Restore of a SolrCloud collection "
+                        + "requires a shared file system mounted at the same path on all nodes!");
+              }
+              snapShooter.validateCreateSnapshot();
+              return snapShooter.createSnapshot();
+            }
+          } catch (Exception exp) {
+            throw new SolrException(
+                SolrException.ErrorCode.SERVER_ERROR,
+                "Failed to backup core=" + coreName + " because " + exp,
+                exp);
+          }
+        });
+  }
+
+  public static class BackupCoreRequestBody extends SolrJerseyResponse {
+
+    @Schema(description = "The name of the repository to be used for backup.")
+    @JsonProperty("repository")
+    public String repository;
+
+    @Schema(description = "The path where the backup will be created")
+    @JsonProperty("location")
+    public String location;
+
+    @JsonProperty("shardBackupId")
+    public String shardBackupId;
+
+    @JsonProperty("prevShardBackupId")
+    public String prevShardBackupId;
+
+    @Schema(
+        description =
+            "The name of the commit which was used while taking a snapshot using the CREATESNAPSHOT command.")
+    @JsonProperty("commitName")
+    public String commitName;
+
+    @Schema(description = "To turn on incremental backup feature")
+    @JsonProperty("incremental")
+    public boolean incremental;
```

Comment:
[-1] We should have this be a `Boolean`, so that BackupCoreAPI has a way to tell whether users actually requested 'false' in their request or whether they just didn't provide a value.  This property is supposed to default to 'true', so it's pretty important to know whether users explicitly requested 'false' or not.

Reply:
I will change it to `Boolean`. Also,
[Q] Should I set default value to 'true' here? Because, and also you mentioned, in `BackupCoreOp.java, If `incremental` is missing they set it to 'true'.